### ISSUE TITLE
math: TBalpha 10.0a4+ 64bit Windows

### ISSUE THREAD

**Issue by Thorin-Oakenpants** [arkenfox/TZP#6]
Info for posterity...

Issue is present
- alpha only (not release) - since 10.0a4
   - e.g. the 9.5.4 stable release (68.12.0esr) is good, but 10.0a4 (68.11.0esr) is not
   - we will wait and see what happens in the first stable release based on ESR78
   - update: it does not happen on TB 10.0 release
- only in 64bit builds
- only on Windows (AFAIK)
- only with TB
   - i.e FF builds with mingw do not produce the issue
- update: just to add to the confusion, I could not reproduce on my Win10 vm

Tested
- FF same release built with and without mingw-64, it does not have the problem

## STR

- bookmark 1: https://arkenfox.github.io/TZP/tests/math.html
- bookmark 2: https://arkenfox.github.io/tzp/tzp.html#math
- start a whole new TB
- go to **site1**
   - note your hash, for me on windows I can just remember `[F3]`
   - ![img1](https://user-images.githubusercontent.com/16656956/92971010-340c7400-f46f-11ea-9ba5-604e060003b8.png)
   - close tab
- go to **site2**
   - note your **1st Edition** hash, you should get a letter, `D` or `G` for Linux, for me I can remember `[B]`
   - ![img2](https://user-images.githubusercontent.com/16656956/92971034-3c64af00-f46f-11ea-9ef1-807b837ee235.png)
   - close tab
- revisit **site1**
   - check the hash or code: it's changed
   - ![img3](https://user-images.githubusercontent.com/16656956/92971050-4686ad80-f46f-11ea-82c2-824a13c6d1d8.png)
   - close tab
- revisit **site2**
   - check the hash or letter: it's changed
   - ![img4](https://user-images.githubusercontent.com/16656956/92971068-4edee880-f46f-11ea-9a30-d3331830e779.png)
   - close tab

**Diffs** (from ~~107~~ 176 items tested)
- The first six items are used on the current TZP test, all items are used in the current math research test
- Note that the last item, `cbrt` is a polyfill

Left: expected. Right: when it changes
![img5](https://user-images.githubusercontent.com/16656956/92971183-7df55a00-f46f-11ea-88c7-77829638c518.png)

**Comment by Thorin-Oakenpants** [comment]
> we will wait and see what happens in the first stable release based on ESR78

It does not happen on TB10.0 - so it seems it is definitely an alpha only issue, and my best guess is it has something to do with `build system` changes

something is/was different in alpha esr78 builds. It will be interesting to see if what happens with the next alpha now release has caught up

here's changes for TB10.0
```
 * Build System
   * Windows + OS X + Linux
     * Bump Go to 1.14.7
     * Bug 31845: Bump GCC version to 9.3.0
     * Bug 34011: Bump clang to 9.0.1
     * Bug 34014: Enable sqlite3 support in Python
     * Bug 34390: Don't copy DBM libraries anymore
     * Bug 34391: Remove unused --enable-signmar option
     * Bug 40004: Adapt Rust project for Firefox 78 ESR [tor-browser-build]
     * Bug 40005: Adapt Node project for Firefox 78 ESR [tor-browser-build]
     * Bug 40006: Adapt cbindgen for Firefox 78 ESR [tor-browser-build]
     * Bug 40037: Move projects over to clang-source [tor-browser-build]
     * Bug 40026: Fix full .mar creation for esr78 [tor-browser-build]
     * Bug 40027: Fix incremental .mar creation for esr78 [tor-browser-build]
     * Bug 40028: Do not reference unset env variables [tor-browser-build]
     * Bug 40031: Add licenses for kcp-go and smux. [tor-browser-build]
     * Bug 40045: Fix complete .mar file creation for dmg2mar [tor-browser-build]
     * Bug 40065: Bump debootstrap-image ubuntu_version to 20.04.1 [tor-browser-build]
     * Bug 40087: Deterministically add HTTPS Everywhere into omni.ja [tor-browser-build]
   * Windows
     * Bug 34230: Update Windows toolchain for Firefox 78 ESR
     * Bug 40015: Use only 64bit fxc2 [tor-browser-build]
     * Bug 40017: Enable stripping again on Windows [tor-browser-build]
     * Bug 40052: Bump NSIS to 3.06.1 [tor-browser-build]
     * Bug 40061: Omit the Windows default browser agent from the build [tor-browser]
     * Bug 40071: Be explicit about no SEH with mingw-w64 on 32bit systems [tor-browser-build]
     * Bug 40077: Don't pass --no-insert-timestamp when building Firefox [tor-browser-build]
     * Bug 40090: NSIS 3.06.1 based builds are not reproducible anymore [tor-browser-build]
   * OS X
     * Bug 34229: Update macOS toolchain for Firefox 78 ESR
     * Bug 40003: Update cctools version for Firefox 78 ESR [tor-browser-build]
     * Bug 40018: Add libtapi project for cctools [tor-browser-build]
     * Bug 40019: Ship our own runtime library for macOS [tor-browser-build]
   * Linux
     * Bug 34359: Adapt abicheck.cc to deal with newer GCC version
     * Bug 34386: Fix up clang compilation on Linux
     * Bug 40053: Also create the langpacks tarball for non-release builds [tor-browser-build]
```

**Comment by Thorin-Oakenpants** [comment]
these are the only diffs in 64 TB 10 (no issue) vs 10.0a7 (issue)
 - resource://gre/chrome/toolkit/content/global/buildconfig.html

![diffs](https://user-images.githubusercontent.com/16656956/93974400-489d1600-fd65-11ea-839e-a3636c176e23.png)

Maybe it's the library versions. IDK. Maybe the next alpha will self-correct. IDK

**Comment by Thorin-Oakenpants** [comment]
> It will be interesting to see if what happens with the next alpha now release has caught up

> Maybe the next alpha will self-correct. IDK

...aaaaaaand in TB 10.5a1 the issue seems to be resolved

**Comment by Thorin-Oakenpants** [comment]
> ...aaaaaaand in TB 10.5a1 the issue seems to be resolved

and I spoke too soon. NOT RESOLVED

**Comment by Thorin-Oakenpants** [comment]
I've tried a few times, but not that much, and haven't replicated this for some time. At one stage I thought the problem had gone away, but then it happened again. Seems intermittent.

Maybe with the move to ESR91.1 together with RFP's math protection this is now resolved. Will leave open for a while, and I promise to test more often - haven't had much time for TB lately, sorry, not sorry :)
